src/breadcrumb-group/item/item.tsx

```diff
@@ -1,19 +1,117 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { BreadcrumbGroupProps, BreadcrumbItemProps } from '../interfaces';
 import InternalIcon from '../../icon/internal';
 import styles from './styles.css.js';
 import clsx from 'clsx';
 import useFocusVisible from '../../internal/hooks/focus-visible';
 import { fireCancelableEvent, isPlainLeftClick } from '../../internal/events';
 import { getEventDetail } from '../internal';
+import { Transition } from '../../internal/components/transition';
+import PopoverContainer from '../../popover/container';
+import PopoverBody from '../../popover/body';
+import Portal from '../../internal/components/portal';
+import popoverStyles from '../../popover/styles.css.js';
+import { useContainerQuery } from '../../internal/hooks/container-queries';
+import { useMergeRefs } from '../../internal/hooks/use-merge-refs';
+
+type BreadcrumbItemWithPopoverProps<T extends BreadcrumbGroupProps.Item> =
+  React.AnchorHTMLAttributes<HTMLAnchorElement> & {
+    item: T;
+  };
+
+const BreadcrumbItemWithPopover = <T extends BreadcrumbGroupProps.Item>({
+  item,
+  ...anchorAttributes
+}: BreadcrumbItemWithPopoverProps<T>) => {
+  const focusVisible = useFocusVisible();
+  const [textTruncated, setTextTruncated] = useState(false);
+  const [openPopover, setOpenPopover] = useState(false);
+  const trackRef = useRef<HTMLElement>(null);
+  const [textWidth, textRef] = useContainerQuery(rect => rect.width, []);
+  const mergedRef = useMergeRefs(trackRef, textRef);
+  const virtualTextRef = useRef<HTMLElement>(null);
+
+  const popoverContent = (
+    <Transition in={openPopover}>
+      {() => (
+        <PopoverContainer
+          trackRef={trackRef}
+          size="small"
+          fixedWidth={false}
+          position="bottom"
+          arrow={position => (
+            <div className={clsx(popoverStyles.arrow, popoverStyles[`arrow-position-${position}`])}>
+              <div className={popoverStyles['arrow-outer']} />
+              <div className={popoverStyles['arrow-inner']} />
+            </div>
+          )}
+        >
+          <PopoverBody dismissButton={false} dismissAriaLabel={undefined} onDismiss={() => {}} header={undefined}>
+            {item.text}
+          </PopoverBody>
+        </PopoverContainer>
+      )}
+    </Transition>
+  );
+
+  useEffect(() => {
+    if (textWidth && virtualTextRef && virtualTextRef.current) {
+      if (Math.round(virtualTextRef.current.clientWidth) > Math.round(textWidth)) {
```

Comment:
Container query uses `getBoundingClientRect` internally. Should `virtualTextRef` use the same method for consistency?